README.md

```diff
@@ -1,2 +1,331 @@
+
+2. During later years, community moderators can propose an "impersonation claim" to governance vote if someone is clearly impersonating or misleading users. 
+
+Names that expire or subject to a claim are moved back into the Farcaster Treasury, which may choose to transfer ownership to someone with "prior ownership" claims or auction them to the highest bidder. For impersonation claims, any fees that were previously paid are forefit.
+
+## 3.3 Recovery
+
+Names and Addresses can be recovered if user loses the keys to the address holding them. Both contracts implement a time-delayed recovery system that allows a **recovery address** to request a transfer to a new address. If the custody address does not cancel the transfer within 3 days, the recovery address can complete the transfer. 
+
+Users can set the recovery address to another address in their wallet, a multi-sig shared with friends, or a third-party recovery service. Users can also change the recovery address at any time. Ownership remains decentralized because the recovery address cannot make a transfer that the custody address does not approve. 
+
+Transferring the asset to a new custody address must unset the recovery address. Otherwise, users may purchase a name on OpenSea only to have the previous owner claim it back stealthily with their recovery address.
+
+# 4. Replication
+
+Replication is the process by which Hubs accept new messages and determine a user's state. 
+
+Users send [messages](#41-self-authenticated-message) to a Hub for every action they take. If a user likes a URL, unlikes it, and likes it again, that creates three messages. A Hub that receives all messages will determine the current state of the URL as *liked by the user*. The Hub discards the first two messages to save space since they are no longer needed. Merging messages at the Hub level avoids client disagreements on state and saves space.
+
+Every message type will have different rules for the merge operation. For example, two likes on the same cast by the same user can be condensed into one, while two replies cannot. Hubs implement a Set for each message type, which is a [conflict-free replicated data type](https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type) that encodes specific validation and merge rules.
+
+Sets ensure [strong eventual consistency](https://en.wikipedia.org/wiki/Eventual_consistency#Strong_eventual_consistency) so that two Hubs that receive the same messages over any period will always reach the same state. This property makes Hubs highly available since they can go offline at any time and always get back into sync. Formally, Sets are anonymous Δ-state CRDTs[^delta-state], and each message is a join-irreducible update on the set. 
+
+<!-- Diagram of all user data types -->
+
+## 4.1 Signers
```

Comment:
Nit: this name leads to think that the signer is a user, not an object. Maybe SignerMessage is more explicit